src/quicklist.c

```diff
@@ -802,8 +808,9 @@ void quicklistReplaceEntry(quicklistIter *iter, quicklistEntry *entry,
     if (likely(!QL_NODE_IS_PLAIN(entry->node) && !isLargeElement(sz, quicklist->fill) &&
         (newentry = lpReplace(entry->node->entry, &entry->zi, data, sz)) != NULL))
     {
+        size_t new_entry_size = zmalloc_usable_size(newentry);
```

Comment:
I think we can add an lpReplaceUsable(...) size_t* newsize, size_t* oldsize), it seems that this can reduce most of the overhead.

Reply:
Done in [0725ef3](https://github.com/redis/redis/pull/14119/commits/0725ef39700db2dcc9496335774d7c8c9addd5ba)